### DISCUSSION TITLE
Dont save specific component

### DISCUSSION THREAD

**Discussion by dingedi** [GrapesJS/grapesjs#3793] [Q&A]
Is there a way to not save an item when saving?

I load a complete HTML page (the layout) and there is only a small part which is editable by the user, I would only like to save this part with the storage manager remote.

thanks in advance :)

**Comment by TheTakylo** [comment]
Same  question it would be really good a guide or clear explanations to be able to display edit a single part of a loaded template

**Reply by dingedi** [reply]
Adding an option would be possible without too much change @artf  ? Or it's a big change in the code. I can try doing a PR if it doesn't require too many changes.

**Comment by artf** [comment]
Here is an example of using components as placeholders https://jsfiddle.net/artur_arseniev/w6mvq4bx/

**Reply by dingedi** [reply]
Thank you, actually the structure I have is a little more complex (and surely more widespread), the editable content is in the layout of my page which should not be editable.

Thanks to your example it seems to work, but unable to get the HTML code (it is empty), what did I do wrong?
[https://jsfiddle.net/hyd249jm/46/](https://jsfiddle.net/hyd249jm/46/)

**Reply by dev-webxdev** [reply]
Same problem for me @artf

**Reply by dingedi** [reply]
It's strange, the HTML code does not appear but in fact the storageManager saves all the data well, even those excluded with the type placeholder.

How can we prevent it from being saved? and only save the editable part?

**Reply by artf** [reply]
HTML does not appear because the root placeholder has this
```js
toHTML() {
  return ''
}
```
you can indeed traverse components and return only the part you need, eg.
```js
toHTML() {
  const editable = this.findType('edt')[0];
  return editable ? editable.toHTML() : '';
}
```
> How can we prevent it from being saved? and only save the editable part?

Well, you skip placeholder's HTML but you have to save them in the project (JSON part), otherwise how would expect to load them back?!

**Reply by dingedi** [reply]
My goal is to display a page generate with my cms and make a div editable with grapejs.

I therefore want to save only the editable part (components, styles, html, css) and not all the layout which is generated by my cms.


It doesn't seem to be working for now  all components and styles are saved while I only want editable ones and no html is generated: / 

[https://jsfiddle.net/u45q8px2/18/](https://jsfiddle.net/u45q8px2/18/)

Thanks a lot for the help

**Reply by artf** [reply]
toHTML example is to apply on `plh` component not `edt`

Anyway, I understand what you want, but you have to store the JSON if you want to allow the user to reopen the same project, otherwise, once saved and reloaded the template you won't see placeholders... do you understand what I mean?

**Comment by dev-webxdev** [comment]
is it possible to save only a specific part with the storage manager? (components, css, html, etc) @artf
